pages/index.js

```diff
@@ -31,15 +33,69 @@ const Index = ({
   const [showResidentForm, setShowResidentForm] = useState(false);
   const [referralCompletion, setReferralCompletion] = useState({ tis: null });
   const [referralSummary, setReferralSummary] = useState([]);
+  const [signpostSummary, setSignpostSummary] = useState([]);
+  const [referrerData, setReferrerData] = useState({
+    'referer-email': refererInfo?.email,
+    'referer-name': refererInfo?.name,
+    'referer-organisation': refererInfo?.iss
+  });
 
   const residentFormCallback = val => {
     setShowResidentForm(val);
   };
   const residentInfoCallback = value => {
     setResidentInfo(value);
   };
+  const updateSignpostSummary = service => {
+    let newSignpostSummary;
+    if (
+      signpostSummary.some(x => x.name == service.name && x.categoryName == service.categoryName)
+    ) {
+      newSignpostSummary = signpostSummary.filter(
+        x => x.name != service.name || x.categoryName != service.categoryName
+      );
+    } else newSignpostSummary = signpostSummary.concat([{ ...service }]);
+    setSignpostSummary(newSignpostSummary);
+    setEmailBody(updateEmailBody(newSignpostSummary));
+  };
```

Comment:
Do I understand this code correctly?:
It's a function that takes in a service,
If within existing signpostSummary array it finds a service with matching data
    --> then it takes all the services with that do not match
Else
    --> adds it in.
    
 Oh... I see... so if we pass in the same service the 2nd time, then it gets removed from the list (sort of like unselected)?
 A comment above line 54 would help tremendously if this is the case. As its not immediately obvious what it does due to all the bool and array expressions. Took me couple minutes to figure out the intent behind the function. 😅

Reply:
Yeah, it might seem a bit confusing, but it's essentially adding and removing relevant services based on them being ticked and unticked